Sources/AST/Environment.swift

```diff
@@ -350,6 +350,16 @@ public struct Environment {
       if binaryExpression.opToken.isBooleanOperator {
         return .basicType(.bool)
       }
+      if binaryExpression.opToken == .dot {
+        switch type(of: binaryExpression.lhs, enclosingType: enclosingType, callerCapabilities: callerCapabilities, scopeContext: scopeContext) {
+        case .arrayType(_):
+          return .basicType(.int)
+        case .fixedSizeArrayType(_):
+          return .basicType(.int)
+        default:
+          break
+        }
+      }
```

Comment:
It's not saying that all arrays are of type `Int`, but instead all array properties are of type `Int`. Since the only property of arrays at this time is `size`, then all properties are of type `Int`.